bittensor/core/async_subtensor.py

```diff
@@ -279,20 +405,41 @@ async def get_hyperparameter(
         block_hash: Optional[str] = None,
         reuse_block: bool = False,
     ) -> Optional[Any]:
-        """
-        Retrieves a specified hyperparameter for a specific subnet.
+        """Retrieves a specified hyperparameter for a specific subnet.
 
-        Arguments:
-            param_name (str): The name of the hyperparameter to retrieve.
+        This method queries the blockchain for subnet-specific hyperparameters such as difficulty, tempo, immunity
+        period, and other network configuration values.
+
+        Args:
+            param_name (str): The name of the hyperparameter to retrieve (e.g., "Difficulty", "Tempo",
+                "ImmunityPeriod").
             netuid (int): The unique identifier of the subnet.
-            block: the block number at which to retrieve the hyperparameter. Do not specify if using block_hash or
-                reuse_block
-            block_hash (Optional[str]): The hash of blockchain block number for the query. Do not specify if using
-                block or reuse_block
-            reuse_block (bool): Whether to reuse the last-used block hash. Do not set if using block_hash or block.
+            block (Optional[int]): The block number at which to retrieve the hyperparameter. Do not specify if using
+                block_hash or reuse_block.
+            block_hash (Optional[str]): The hash of the blockchain block for the query. Do not specify if using block or
+                reuse_block.
+            reuse_block: Whether to reuse the last-used block hash. Do not set if using block_hash or block.
 
         Returns:
-            The value of the specified hyperparameter if the subnet exists, or None
+            The value of the specified hyperparameter if the subnet exists, None otherwise.
+
+        Example:
+            # Get difficulty for subnet 1
+            difficulty = await subtensor.get_hyperparameter("Difficulty", netuid=1)
+
+            # Get tempo at a specific block
+            tempo = await subtensor.get_hyperparameter(
+                "Tempo",
```

Comment:
```suggestion
                param_name="Tempo",
```
Let's be as transparent as possible in the docstring's examples and use all arguments to pass values ​​to parameters. This will be very useful when we parse examples for the MCP server.